packages/context/test/unit/value-promise.test.ts

```diff
@@ -55,7 +55,7 @@ describe('getDeepProperty', () => {
 
   it('gets the root value with a path', () => {
     const obj = {x: {y: 1}};
-    expect(getDeepProperty(obj, 'x.y')).to.eql(1);
+    expect(getDeepProperty<number>(obj, 'x.y')).to.eql(1);
```

Comment:
I think the changes in this files should not be strictly required, because `getDeepProperty` defaults to `any` type, which is fine as long as `expect` is concerned.

Are you adding `<number>` specialization to verify that `getDeepProperty` accepts a generic parameter? We should write a new test to do so, one that will also verify at compile time that the result of `getDeepProperty` is indeed the type we specified. (Most of) the existing tests should remain unchanged, that way we can verify that `getDeepProperty` can be used without the generic parameter too.

 See https://github.com/strongloop/loopback-next/pull/1169 for an example:

https://github.com/strongloop/loopback-next/blob/7ba261850603fecd4d6c16329b47dee63b7daa9d/packages/context/test/acceptance/creating-and-resolving-bindings.ts#L48-L55

To make the intent easier to understand without the need for excessive comments, I would write a helper along the following lines:

```ts
// usage
      it('allows access to a deep property', async () => {
        const key = new BindingKey<object>('foo');
        ctx.bind(key).to({rest: {port: 80}});
        const value = await ctx.get(key.deepProperty<number>('rest.port'));
        expectNumberAtCompileTime(value, 80);
      });

// impl
function expectNumberAtCompileTime(actual: number, expected: number) {
  expect(actual).to.be.a.Number()
  expect(actual).to.equal(expected);
}
```

Reply:
Fixed.